File: testdocs/hello.fs
module HelloWorld

let greet name = sprintf "Hello, %s!" name

[<EntryPoint>]
let main _ =
    printfn "%s" (greet "World")
    0

// Indent/fold fixture. F# ships neither an indents.scm nor a folds.scm, so this
// exercises the indent fallback on both axes. See openspec TEST_PLAN, change
// align-treesitter-providers.
//
// fsautocomplete IS installed now, but it cannot answer for this file: a bare
// .fs outside any project is never added to its loaded projects, so requests
// fail with "Couldn't find <path> in LoadedProjects" and opening the file alone
// logs an UnhandledPromiseRejection. That is fine for the indent/fold cases,
// which need no server response. Cases that do need one use hello.fsx.

type Shape =
    | Circle of float
    | Rect of float * float

let area shape =
    match shape with
    | Circle r ->
        System.Math.PI * r * r
    | Rect (w, h) ->
        w * h

let describe shape =
    let a = area shape
    if a > 100.0 then
        "large"
    elif a > 10.0 then
        "medium"
    else
        "small"

let summarise shapes =
    shapes
    |> List.map area
    |> List.sum
    |> printfn "total: %f"

File: testdocs/hello.hs
module Main where

greet :: String -> String
greet name = "Hello, " ++ name ++ "!"

main :: IO ()
main = putStrLn (greet "World")

-- Indent/fold fixture: the definitions above are one-liners with nothing to
-- fold, which reads as "folding is broken" when it is the file that is thin.
-- See openspec TEST_PLAN, change align-treesitter-providers, AT.2/AT.8.

data Shape
  = Circle Double
  | Rect Double Double
  deriving (Show, Eq)

area :: Shape -> Double
area shape =
  case shape of
    Circle r ->
      pi * r * r
    Rect w h ->
      w * h

describe :: Shape -> String
describe s
  | area s > 100 = "large"
  | area s > 10  = "medium"
  | otherwise    = "small"

summarise :: [Shape] -> IO ()
summarise shapes = do
  let total = sum (map area shapes)
  mapM_ (putStrLn . describe) shapes
  print total
